e2e-tests/run.sh

```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+chmod +x ./e2e-tests/batch-mode-tests.sh
```

Comment:
What is the benefit of this `run.sh`? Can't we directly run `e2e-tests/batch-mode-tests.sh` in `.travis.yml`. Also as I said, I think you can set these files to be executable and avoid the `chmod`.